dissect/target/filesystem.py

```diff
@@ -657,10 +695,14 @@ def readlink_ext(self) -> FilesystemEntry:
         # Default behavior, resolve link own filesystem.
         return fsutil.resolve_link(fs=self.fs, entry=self)
 
-    def stat(self) -> fsutil.stat_result:
-        """Determine the stat information of this entry, resolving any symlinks.
+    def stat(self, follow_symlinks: bool = True) -> fsutil.stat_result:
```

Comment:
The code for most `stat()`'s is also very similar (resolve, do lstat). Is this something we may want to optimize in the base `FilesystemEntry` class?

Reply:
I have also been looking into that. However I decided not to do this as I think `FilesystemEntry` should mostly be abstract and also that when implementing your own subclass you should be triggerd to really think about how these functions that now raise `NotImplementedError`s should be implemented.